dockers/docker-fpm-frr/base_image_files/prefix_list

```
@@ -0,0 +1,180 @@
+# Function to handle prefix list operations for a single ASIC
+handle_prefix_list_single() {
+    local operation=$1
+    local PREFIX_TYPE=$2
+    local network=$3
+
+    if [ $operation == 'status' ] ; then
+        echo "Current prefix lists:"
+        sonic-cfggen -d -v PREFIX_LIST
+    else
+        if [ $operation == 'add' ]; then
+            local prefix_list_entry="{\"PREFIX_LIST\":{\"$PREFIX_TYPE|$network\":{}}}"
+            sonic-cfggen -a "$prefix_list_entry" -w
+            logger -t $operation -p user.info "Added prefix list: $PREFIX_TYPE with network: $network"
+            echo "Added prefix list: $PREFIX_TYPE with network: $network"
+        elif [ $operation == 'remove' ]; then
+            sonic-db-cli CONFIG_DB DEL "PREFIX_LIST|$PREFIX_TYPE|$network"
+            logger -t $operation -p user.info "Removed prefix list: $PREFIX_TYPE with network: $network"
+            echo "Removed prefix list: $PREFIX_TYPE with network: $network"
+        fi
+    fi
+}
+
+prefix_list_operations=("add" "remove" "status")
+supported_prefix_types=("ANCHOR_PREFIX")
+# Main script execution
+if [[ "$1" == "-h" || "$1" == "--help" ]]; then
+    display_help
+fi
+
+check_root_privileges
+check_spine_router
+skip_chassis_supervisor
+
+validate_operation $1 $2
+
+# Read SONiC immutable variables
+[ -f /etc/sonic/sonic-environment ] && . /etc/sonic/sonic-environment
+
+PLATFORM=${PLATFORM:-`sonic-cfggen -H -v DEVICE_METADATA.localhost.platform`}
+
+# Parse the device specific asic conf file, if it exists
+ASIC_CONF=/usr/share/sonic/device/$PLATFORM/asic.conf
+[ -f $ASIC_CONF ] && . $ASIC_CONF
+
+if [[ ($NUM_ASIC -gt 1) ]]; then
+    asic=0
+    while [ $asic -lt $NUM_ASIC ]
+    do        
+        sub_role=`sonic-cfggen -d -v "DEVICE_METADATA['localhost']['sub_role']" -n asic$asic`
+        subtype=`sonic-cfggen -d -v "DEVICE_METADATA['localhost']['subtype']" -n asic$asic`
+        if [ $sub_role == 'FrontEnd' ] ; then
+            if [ $subtype == 'UpstreamLC' ] ; then
```

May be throw error if its not upstreamLC instead of silently ignoring it?

---

I have added the UpstreamLC check under check_spine_router function along with the stderr